File: solutions/csharp/protein-translation/1/ProteinTranslation.cs
using System;
using System.Collections.Generic;

public static class ProteinTranslation
{
    public static string[] Proteins(string strand)
    {
        Dictionary<string, string> proteins = new Dictionary<string, string>();
        proteins.Add("AUG", "Methionine");
        proteins.Add("UUU", "Phenylalanine");
        proteins.Add("UUC",	"Phenylalanine");
        proteins.Add("UUA", "Leucine");
        proteins.Add("UUG", "Leucine");
        proteins.Add("UCU", "Serine");
        proteins.Add("UCC", "Serine");
        proteins.Add("UCA", "Serine");
        proteins.Add("UCG", "Serine");
        proteins.Add("UAU", "Tyrosine");
        proteins.Add("UAC", "Tyrosine");
        proteins.Add("UGU", "Cysteine");
        proteins.Add("UGC", "Cysteine");
        proteins.Add("UGG", "Tryptophan");
        List<string> result = new List<string>();
        for (int i = 0; i < strand.Length; i += 3) {
            string protein = strand.Substring(i, 3);
            if (proteins.ContainsKey(protein)) {
                result.Add(proteins[protein]);
            } else {
                break;
            }
        }
        return result.ToArray();
    }
}

File: solutions/python/protein-translation/1/protein_translation.py
def proteins(strand):
    proteins_dict: dict = {
        "AUG": "Methionine",
        "UUU": "Phenylalanine",
        "UUC": "Phenylalanine",
        "UUA": "Leucine",
        "UUG": "Leucine",
        "UCU": "Serine",
        "UCC": "Serine",
        "UCA": "Serine",
        "UCG": "Serine",
        "UAU": "Tyrosine",
        "UAC": "Tyrosine",
        "UGU": "Cysteine",
        "UGC": "Cysteine",
        "UGG": "Tryptophan",
    }
    result = []
    for i in range(0, len(strand), 3):
        protein = strand[i:i+3]
        if protein in proteins_dict:
            result.append(proteins_dict[protein])
        else:
            break
    return result